Code/GraphMol/Fingerprints/Wrap/MHFPWrapper.cpp

```diff
@@ -0,0 +1,316 @@
+      bool rings = true,
+      bool isomeric = false,
+      bool kekulize = true,
+      unsigned char min_radius = 1) {
+  return mhfpEnc->Encode(mol, radius, rings, isomeric, kekulize, min_radius);
+}
+
+VectMinHashVect
+EncodeSmilesBulk(MHFPEncoder* mhfpEnc,
+         python::list& smiles,
+         unsigned char radius = 3,
+         bool rings = true,
+         bool isomeric = false,
+         bool kekulize = true,
+         unsigned char min_radius = 1) {
+  std::vector<std::string> vec = ListToVector<std::string>(smiles);
+  return mhfpEnc->Encode(vec, radius, rings, isomeric, kekulize, min_radius);
+}
+
+
+// There are access problems for vector_indexing_suite and std::vector<ROMol>.
+// So let's fallback to a unefficient Python list. 
+VectMinHashVect
+EncodeMolsBulk(MHFPEncoder* mhfpEnc,
+         python::list& mols,
+         unsigned char radius = 3,
+         bool rings = true,
+         bool isomeric = false,
+         bool kekulize = true,
+         unsigned char min_radius = 1) {
+  std::vector<ROMol> vec = ListToVector<ROMol>(mols);
+  return mhfpEnc->Encode(vec, radius, rings, isomeric, kekulize, min_radius);
+}
+
+// SECFP
+
+ExplicitBitVect
+EncodeSECFPSmiles(MHFPEncoder* mhfpEnc,
+          std::string smiles,
+          unsigned char radius = 3,
+          bool rings = true,
+          bool isomeric = false,
+          bool kekulize = true,
+          unsigned char min_radius = 1,
+          size_t length = 2048) {
+  return mhfpEnc->EncodeSECFP(smiles, radius, rings, isomeric, kekulize, min_radius, length);
+}
+
+ExplicitBitVect
+EncodeSECFPMol(MHFPEncoder* mhfpEnc,
+         ROMol mol,
+         unsigned char radius = 3,
+         bool rings = true,
+         bool isomeric = false,
+         bool kekulize = true,
+         unsigned char min_radius = 1,
+         size_t length = 2048) {
+  return mhfpEnc->EncodeSECFP(mol, radius, rings, isomeric, kekulize, min_radius, length);
+}
+
+VectExplicitBitVect
+EncodeSECFPSmilesBulk(MHFPEncoder* mhfpEnc,
+            python::list& smiles,
+            unsigned char radius = 3,
+            bool rings = true,
+            bool isomeric = false,
+            bool kekulize = true,
+            unsigned char min_radius = 1,
+            size_t length = 2048) {
+  std::vector<std::string> vec = ListToVector<std::string>(smiles);
+  return mhfpEnc->EncodeSECFP(vec, radius, rings, isomeric, kekulize, min_radius, length);
+}
+
+VectExplicitBitVect
+EncodeSECFPMolsBulk(MHFPEncoder* mhfpEnc,
+          python::list& mols,
+          unsigned char radius = 3,
+          bool rings = true,
+          bool isomeric = false,
+          bool kekulize = true,
+          unsigned char min_radius = 1,
+          size_t length = 2048) {
+  std::vector<ROMol> vec = ListToVector<ROMol>(mols);
+  return mhfpEnc->EncodeSECFP(vec, radius, rings, isomeric, kekulize, min_radius, length);
+}
+
+BOOST_PYTHON_FUNCTION_OVERLOADS(CreateShinglingFromSmilesOverloads, CreateShinglingFromSmiles, 2, 7)
```

Comment:
Nice! I haven't looked at this macro in a long time, but it makes things tons easier!